src/Common/OvercommitTracker.cpp

```diff
@@ -22,6 +24,9 @@ bool OvercommitTracker::needToStopQuery(MemoryTracker * tracker)
 {
     std::unique_lock<std::mutex> lk(overcommit_m);
 
+    if (max_wait_time == ZERO_MICROSEC)
```

Comment:
Maybe it make sense to make it atomic?

Reply:
Read of this variable happens always when `overcommit_m` is acquired, because it's used in `wait_for` call. So I think it doesn't.